packages/client/src/components/app/Layout.svelte

```diff
@@ -282,7 +282,14 @@
         </div>
       </div>
     {/if}
-    <div class="main-wrapper">
+    <div
+      class="main-wrapper"
+      on:click={() => {
+        if ($builderStore.inBuilder) {
+          builderStore.actions.selectComponent($builderStore.screen._id)
```

Comment:
This should be the `screenId` variable that we already have above. The correct ID for the screen being selected is the screen ID suffixed by `-screen`. This still works because the builder thinks this is invalid so will actually redirect to the correct ID, but it caused a double URL update so would be better to update this to send `screenId` instead